python/bind_slice_methods.h

```diff
@@ -82,8 +83,43 @@ template <class T> struct slicer {
     return self.slice(Slice(dim, i));
   }
 
+  static auto get_by_value(T &self,
+                           const std::tuple<Dim, VariableConstView> &value) {
+    auto [dim, i] = value;
+    return slice(self, dim, i);
+  }
+
   static auto get_range(T &self,
                         const std::tuple<Dim, const py::slice> &index) {
+    auto [dim, py_slice] = index;
+    if constexpr (std::is_same_v<T, DataArray> ||
+                  std::is_same_v<T, DataArrayView>) {
+      try {
+        auto step = py::getattr(py_slice, "step");
+        if (step.is_none()) {
+          std::runtime_error(
+              "Step cannot be specified for value based slicing.");
+        }
+        auto start = py::getattr(py_slice, "start");
+        auto stop = py::getattr(py_slice, "stop");
+        auto start_var =
+            start.is_none()
+                ? Variable{}
+                : py::getattr(py_slice, "start").cast<VariableConstView>();
+        auto stop_var =
+            stop.is_none()
+                ? Variable{}
+                : py::getattr(py_slice, "stop").cast<VariableConstView>();
+
+        if (stop_var || start_var) {
+          // In the case of ['dim', :] want to default to index based to avoid
+          // unit clashes.
```

Comment:
This should not actually be necessary? Doesn't `slice()` handle this correctly when passed two invalid `VariableConstView` for start and end?

Reply:
This is probably now unnecessary yes I shall remove.